### PR TITLE
Improve development flow with skaffold

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR improves the development flow on the dogfooding cluster when it's needed to continuously update the dashboard on a remote cluster to review or test any changes. Skaffold is able to automate all steps needed, like building a new image, pushing the image to the repo, and then updating the CR.

[Using scaffold on dogfooding cluster](https://www.youtube.com/watch?v=MeXLKcX-mmUE)

https://youtu.be/MeXLKcX-mmU

### What issues does this PR fix or reference?

resolves https://github.com/eclipse/che/issues/22116

### Is it tested? How?

1. Create a new workspace using this link: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#https://github.com/eclipse-che/che-dashboard/tree/skaffold
2. Log in to the remote cluster with Eclipse Che deployed.
3. Log in to the container registry you want to use.
4. ```sh
	export DEFAULT_REPO=<your-registry>/<project-id>
	```
5. Do some changes in the `che-dashboard` project, and run `yarn build`.
6. Run 
	```sh
	scaffold run --cleanup=false --default-repo=$DEFAULT_REPO
	```
	Skaffold builds a new image, pushes it to the registry, and patches the CR.
7. Watch changes you've made in the dashboard on the remote cluster.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -195,6 +196,35 @@ To a new image and apply it to the CheCluster, run:
...
 
+### Update dashboard on remote cluster using `skaffold.dev`
+
+To update the dashboard deployment you need access to the Kubernetes cluster (see [Running locally against remote Che Cluster (Node.js v.16)](#running-locally-against-remote-che-cluster-nodejs-v16))
+
+Then proceed with the following steps: 
+
+```sh
+# export an environment variable to define a repository you want images to be pushed, e.g.:
+export DEFAULT_REPO=quay.io/<project-id>
```
Comment (olexii4) [inline README.md:207]: I suggest using existing environment variables:
https://github.com/eclipse-che/che-dashboard/blob/main/run/build-and-patch.sh#L16
```
DEFAULT_REPO="${IMAGE_REGISTRY_HOST}/${IMAGE_REGISTRY_USER_NAME}"
```
WDYT?

File: skaffold.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  name: che-dashboard
+build:
+  tagPolicy:
+    sha256: {}
+  artifacts:
+    - image: che-dashboard
+      context: .
+      custom:
+        buildCommand: |
+          podman build  --tag=$IMAGE -f build/dockerfiles/skaffold.Dockerfile . && podman push $IMAGE
```
Comment (olexii4) [inline skaffold.yaml:13]: Could you prepare a script, like this one ```che-dashboard/scripts
/dash-licenses.sh```, which helps to choose automatically what to use **Docker** or **Podman**?

File: devfile.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ metadata:
 components:
   - name: tools
     container:
-      image: quay.io/devfile/universal-developer-image:ubi8-38da5c2
+      image: quay.io/okurinny/dashboard.dev:skaffold
```
Comment (olexii4) [inline devfile.yaml:8]: If we are going to support our own image for development we should prepare a separate repository, like this one https://github.com/che-incubator/dash-licenses with a Dockerfile and workflows.

File: README.md (hunk 1 above)
Comment (akurinnoy) [reply README.md:207]: done

File: skaffold.yaml (hunk 2 above)
Comment (akurinnoy) [reply skaffold.yaml:13]: done

File: devfile.yaml (hunk 3 above)
Comment (akurinnoy) [reply devfile.yaml:8]: done: https://github.com/akurinnoy/dashboard.dev

**Review by olexii4 — APPROVED** [review]

**Review by ibuziuk — APPROVED** [review]

**Pull request merged**